barretenberg/cpp/pil/avm/avm_kernel.pil

```diff
@@ -27,15 +27,16 @@ namespace avm_kernel(256);
     // FIXED INDEXES
     // Exists checks
     pol START_NOTE_HASH_EXISTS_WRITE_OFFSET = 0;
-    pol START_NULLIFIER_EXISTS_OFFSET = 32; // START_NOTE_HASH_EXISTS_WRITE_OFFSET + MAX_NOTE_HASH_READ_REQUESTS_PER_CALL TODO: exists and non exists
+    pol START_NULLIFIER_EXISTS_OFFSET = 32; // START_NOTE_HASH_EXISTS_WRITE_OFFSET + MAX_NOTE_HASH_READ_REQUESTS_PER_CALL
+    pol START_NULLIFIER_NON_EXISTS_OFFSET = 64; // START_NULLIFIER_EXISTS_OFFSET +  MAX_NULLIFIER_READ_REQUESTS_PER_CALL
     pol START_L1_TO_L2_MSG_EXISTS_WRITE_OFFSET = 96; //  START_NULLIFIER_EXISTS_OFFET + (MAX_NULLIFIER_READ_REQUESTS_PER_CALL + MAX_NULLIFIER_NON_EXISTENT_READ_REQUESTS_PER_CALL)
 
     // Public storage requests
-    pol START_SLOAD_WRITE_OFFSET = 112; // START_L1_TO_L2_MSG_EXISTS_WRITE_OFFSET + MAX_L1_TO_L2_MSG_READ_REQUESTS_PER_CALL
-    pol START_SSTORE_WRITE_OFFSET = 144; // START_SSTORE_WRITE_OFFSET + MAX_PUBLIC_DATA_UPDATE_REQUESTS_PER_CALL
+    pol START_SSTORE_WRITE_OFFSET = 128; // START_L1_TO_L2_MSG_EXISTS_WRITE_OFFSET  + MAX_PUBLIC_DATA_UPDATE_REQUESTS_PER_CALL
+    pol START_SLOAD_WRITE_OFFSET = 144; // START_SSTORE_WRITE_OFFSET  + MAX_L1_TO_L2_MSG_READ_REQUESTS_PER_CALL
     
     // Emit data 
-    pol START_EMIT_NOTE_HASH_WRITE_OFFSET = 160; // START_SLOAD_WRITE_OFFSET + MAX_PUBLIC_DATA_READS_PER_CALL
+    pol START_EMIT_NOTE_HASH_WRITE_OFFSET = 160; // START_SSTORE_WRITE_OFFSET + MAX_PUBLIC_DATA_READS_PER_CALL
     pol START_EMIT_NULLIFIER_WRITE_OFFSET = 176; // START_EMIT_NOTE_HASH_WRITE_OFFSET + MAX_NEW_NOTE_HASHES_PER_CALL
```

Comment:
Is the comment right for start_emit? if sload comes after sstore, why are we offsetting with sstore here?

Reply:
`avm_kernel_trace.hpp` has `static const uint32_t START_EMIT_NOTE_HASH_WRITE_OFFSET = START_SLOAD_WRITE_OFFSET + MAX_PUBLIC_DATA_READS_PER_CALL;`, so i think the comment for EMIT_NOTE_HASH offset is wrong here.